src/spatial/models/heatmapModel.py

```
@@ -0,0 +1,799 @@
+            )
+            grouped_df["pm25"] = grouped_df["pm2_5"].round(2)
+
+            self.gdf = gpd.GeoDataFrame(
+                grouped_df,
+                geometry=gpd.points_from_xy(grouped_df["longitude"], grouped_df["latitude"]),
+                crs="EPSG:4326",
+            )
+            self.logger.info("Created aggregated GeoDataFrame with %d sites.", len(self.gdf))
+            return True
+
+        except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)